tee-worker/ts-tests/integration-tests/data-providers.test.ts

```diff
@@ -0,0 +1,140 @@
+        const eventsPromise = subscribeToEventsWithExtHash(reqExtHash, context);
+        try {
+            // CLIENT = "$CLIENT_BIN -p $NPORT -P $WORKER1PORT -u $NODEURL -U $WORKER1URL"
+            const commandPromise = zx`${clientDir} -p ${nodePort} -P ${workerPort} -u ${
+                nodeProtocal + nodeHostname
+            } -U ${workerProtocal + workerHostname}\
+                  trusted -d link-identity did:litentry:substrate:${formatAddress}\
+                  did:${credentialDefinitions.mockDid}\
+                  ${credentialDefinitions.mockWeb3Network}`;
+
+            await commandPromise;
+        } catch (error: any) {
+            console.log(`Exit code: ${error.exitCode}`);
+            console.log(`Error: ${error.stderr}`);
+            throw error;
+        }
+
+        const events = (await eventsPromise).map(({ event }) => event);
+        assert.equal(events.length, 1);
+    }
+
+    async function requestVc(id: string, index: number) {
+        const credentialDefinitions = vip3CredentialJson.find((item) => item.id === id) as CredentialDefinition;
+
+        const assertion = {
+            [credentialDefinitions.assertion.id]: credentialDefinitions.assertion.payload,
+        };
+
+        let currentNonce = (await getSidechainNonce(context, teeShieldingKey, substrateIdentities[index])).toNumber();
+        const getNextNonce = () => currentNonce++;
+        const nonce = getNextNonce();
+        console.log(nonce, substrateIdentities[index].toHuman(), u8aToHex(keyringPairs[index].publicKey));
+
+        const requestIdentifier = `0x${randomBytes(32).toString('hex')}`;
+        const requestVcCall = await createSignedTrustedCallRequestVc(
+            context.api,
+            context.mrEnclave,
+            context.api.createType('Index', nonce),
+            new PolkadotSigner(keyringPairs[index]),
+            substrateIdentities[index],
+            context.api.createType('Assertion', assertion).toHex(),
+            context.api.createType('Option<RequestAesKey>', aesKey).toHex(),
+            requestIdentifier
+        );
+        const res = await sendRequestFromTrustedCall(context, teeShieldingKey, requestVcCall);
+        await assertIsInSidechainBlock(`${Object.keys(assertion)[0]} requestVcCall`, res);
+
+        const vcResults = context.api.createType('RequestVCResult', res.value) as unknown as RequestVCResult;
+        const decryptVcPayload = decryptWithAes(aesKey, vcResults.vc_payload, 'utf-8').replace('0x', '');
+        const vcPayloadJson = JSON.parse(decryptVcPayload);
+
+        assert.equal(vcPayloadJson.credentialSubject.values[0], credentialDefinitions.expectedCredentialValue);
+    }
+
+    if (argvId) {
+        const credentialDefinitions = vip3CredentialJson.find((item) => item.id === argvId) as CredentialDefinition;
+        if (credentialDefinitions) {
+            const credentialDefinitions = vip3CredentialJson.find((item) => item.id === argvId) as CredentialDefinition;
```

Comment:
I've read this lines couple of times and it seems there are some either duplication or wrong naming. We should not be able to redefine const variable 🤔 @Verin1005 can you check it

Reply:
Thanks. Removed!